### PR TITLE
Issue #co-267 fix: Request traceability with proper logs

### PR DESCRIPTION
**Pull request by vaivk369**
<table>
<td>
  <a href="https://project-sunbird.atlassian.net/browse/CO-267" title="CO-267" target="_blank">Minor-EnhancementCO-267</a>  Request traceability with proper logs (Part 2)
  </td></table>
  <br />
 

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

### Type of change

Please choose appropriate options.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

### How Has This Been Tested?

Please describe the tests that you ran to verify your changes in the below checkboxes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Ran Test A
- [ ] Ran Test B

**Test Configuration**:
* Software versions: Nodejs 8.11.2
* Hardware versions:

### Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: src/service/programService.js [hunk 1]
```diff
@@ -31,7 +32,7 @@ var async = require('async')
 
 const queryRes_Max = 1000;
 const queryRes_Min = 300;
-const stackTrace_MaxLimit = 500;
+//const stackTrace_MaxLimit = 500;
```
Comment (vinukumar-vs) [inline src/service/programService.js:35]: Why did you removed this? We should not have very big log message..

Comment (vinukumar-vs) [reply src/service/programService.js:35]: May be you can update the value if required for errors.

Comment (vaivk369) [reply src/service/programService.js:35]: @vinukumar-vs This variable was not being used in this file. That is why removed.

File: src/middlewares/request.middleware.js [hunk 2]
```diff
@@ -45,12 +45,13 @@ function createAndValidateRequestBody (req, res, next) {
   req.body.url = req.url
   req.body.path = req.route.path
   req.body.params = req.body.params ? req.body.params : {}
-  req.body.params.msgid = req.get('x-msgid') || req.body.params.msgid || uuidV1()
+  req.body.params.msgid = req.get('x-request-id') || req.get('x-msgid') || req.body.params.msgid || uuidV1()
```
Comment (vinukumar-vs) [inline src/middlewares/request.middleware.js:48]: Add new porperty to the prarams "resmsgid"
```
req.body.params.resmsgid = req.body.params.resmsgid || req.get('x-request-id') || req.body.params.msgid || uuid()
```

File: src/service/bulkJobService.js [hunk 3]
```diff
@@ -207,7 +207,7 @@ function errorResponse(data,errCode) {
   response.id = data.apiId
   response.ver = data.apiVersion
   response.ts = new Date()
-  response.params = getParams(data.msgId, 'failed', data.errCode, data.errMsg)
+  response.params = getParams(data.msgid, 'failed', data.errCode, data.errMsg)
```
Comment (vinukumar-vs) [inline src/service/bulkJobService.js:210]: Pass resmsgid as one more paramerter with default=null

File: src/helpers/responseUtil.js [hunk 4]
```diff
@@ -20,15 +20,15 @@ const errorResponse = (data,errCode) => {
...
   response.ver = data.apiVersion
   response.ts = new Date()
-  response.params = getParams(data.msgId, 'failed', data.errCode, data.errMsg)
+  response.params = getParams(data.msgid, 'failed', data.errCode, data.errMsg)
   response.responseCode = errCode+'_'+data.responseCode
   response.result = data.result
   return response
 }
 
 const getParams = (msgId, status, errCode, msg) => {
```
Comment (vinukumar-vs) [inline src/helpers/responseUtil.js:29]: Pass resmsgid as one more paramerter with default=null

**Review by vinukumar-vs — CHANGES_REQUESTED** [review]

**Review by vinukumar-vs — APPROVED** [review]

**Pull request merged**